src/Playwright/Core/BrowserContext.cs

```diff
@@ -325,11 +327,12 @@ internal void OnRoute(Route route, IRequest request)
             foreach (var item in _routes)
             {
                 if (
-                    (item.Url != null && UrlMatches(request.Url, item.Url)) ||
+                    ((item.Times ?? 0) == 0 || item.HandledCount >= item.Times) &&
```

Comment:
I would move this (line 330-333) into an own class like we did in JavaScript. Because it's the same logic as in the Page. Then it exposes a Match function and a Handle function.

Reply:
I will likely be doing that once I clean up the channels. For now, keeping as is.